ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/AttestationUtil.java

```diff
@@ -137,10 +163,153 @@ public static Bytes32 getAttestationMessageToSign(AttestationData attestationDat
+    checkArgument((bit_0_indices.size() + bit_1_indices.size()) <= MAX_INDICES_PER_ATTESTATION, "AttestationUtil.validate_indexed_attestation: Verify max number of indices");
+    checkArgument(intersection(bit_0_indices, bit_1_indices).size() == 0, "AttestationUtil.validate_indexed_attestation: Verify index sets are disjoint");
+
+    //Verify indices are sorted
+    List<Integer> bit_0_indices_sorted = new ArrayList<Integer>(bit_0_indices);
+    Collections.sort(bit_0_indices_sorted);
+    List<Integer> bit_1_indices_sorted = new ArrayList<Integer>(bit_1_indices);
+    Collections.sort(bit_1_indices_sorted);
+    checkArgument(bit_0_indices.equals(bit_0_indices_sorted) && bit_1_indices.equals(bit_1_indices_sorted));
+
+    List<Validator> validators = state.getValidator_registry();
+    List<BLSPublicKey> pubkeys = Arrays.stream(validators.subList(0, bit_0_indices.size()).toArray())
+            .map((validator) -> {return ((Validator)validator).getPubkey();}).collect(Collectors.toList());
+    pubkeys.addAll(Arrays.stream(validators.subList(0, bit_1_indices.size()).toArray())
+            .map((validator) -> {return ((Validator)validator).getPubkey();}).collect(Collectors.toList()));
+
+    List<Bytes32> message_hashes = new ArrayList<Bytes32>();
+    new AttestationDataAndCustodyBit(indexed_attestation.getData(), false);
+    message_hashes.add(new AttestationDataAndCustodyBit(indexed_attestation.getData(), false).hash_tree_root());
+    message_hashes.add(new AttestationDataAndCustodyBit(indexed_attestation.getData(), true).hash_tree_root());
+
+    BLSSignature signature = indexed_attestation.getSignature();
+    UnsignedLong domain = UnsignedLong.valueOf(get_domain(state, DOMAIN_ATTESTATION, indexed_attestation.getData().getTarget_epoch()));
+    //Verify aggregate signature
+    checkArgument(BLSVerify.bls_verify_multiple(pubkeys, message_hashes, signature, domain));
+  }
+
+  /**
+   * Returns the data slot for the provided AttestationData
+   *
+   * @param state
+   * @param data
+   * @return
+   *
+   * @see <a>https://github.com/ethereum/eth2.0-specs/blob/v0.7.1/specs/core/0_beacon-chain.md#get_attestation_data_slot</a>
+   */
+  public static UnsignedLong get_attestation_data_slot(BeaconState state, AttestationData data) {
+    UnsignedLong committee_count = get_epoch_committee_count(state, data.getTarget_epoch());
+    UnsignedLong offset = (data.getCrosslink().getShard()
+            .plus(UnsignedLong.valueOf(SHARD_COUNT))
+            .minus(get_epoch_start_shard(state, data.getTarget_epoch())))
+              .mod(UnsignedLong.valueOf(SHARD_COUNT));
+    UnsignedLong numerator = get_epoch_start_slot(data.getTarget_epoch()).plus(offset);
+    UnsignedLong denominator = committee_count.dividedBy(UnsignedLong.valueOf(SLOTS_PER_EPOCH));
+    UnsignedLong data_slot = numerator.dividedBy(denominator);
```

Comment:
The order of operations here is wrong. The `numerator` is just `offset`, and then get_epoch_start_slot(...) should be added to the quotient at the end.